### ISSUE TITLE
Last item not fully visible when scrolling in time when using portalId in scrollable div

### ISSUE THREAD

**Issue by 1Jesper1** [Hacker0x01/react-datepicker#4290]
**Describe the bug**
Last item not fully visible when scrolling in time when using portalId in scrollable div

**To Reproduce**

```
() => {
  const [startDate, setStartDate] = useState(new Date('02-02=2023 06:00'));
  return (
  <div>
  <div id='test'></div>
    <DatePicker
      portalId={'test'}
      selected={startDate}
      onChange={(date) => setStartDate(date)}
      showTimeSelect
    />
     </div>
  );
};
```

**Expected behavior**
Should show the full height of last time item.

**Screenshots**
![image](https://github.com/Hacker0x01/react-datepicker/assets/4539406/ac98cad9-d0dc-4c35-87e4-9fc8d419dfff)

**Desktop (please complete the following information):**

- OS: Windows
- BrowserChrome
- Version react-datepicker 4.10.0

**Comment by balajis-qb** [comment]
Hi @1Jesper1 , 
I couldn't able to reproduce the error.  I can able to see the time list fully

![time list scroll issue](https://github.com/Hacker0x01/react-datepicker/assets/145029139/3a516ebf-5351-42c7-938f-8676ee3cfcaa)

**Comment by 1Jesper1** [comment]
@balajis-qb 
Please try:
```
() => {
  const [startDate, setStartDate] = useState(new Date('02-02=2023 11:30 pm'));
  return (
  <div>
  <div id='test'></div>
    <DatePicker
      portalId={'test'}
      selected={startDate}
      onChange={(date) => setStartDate(date)}
      showTimeSelect
    />
     </div>
  );
};
```

**Comment by 1Jesper1** [comment]
![image](https://github.com/Hacker0x01/react-datepicker/assets/4539406/08df95c1-eab6-43eb-889b-50da3baeb2b8)
Also have the bug in Firefox.

**Comment by balajis-qb** [comment]
Yeah, I got the issue.  Initially I misunderstood the issue.  You are mentioning about the initial view right? when we open the date picker with the last time item as selected, in that case we have to manually scroll to the selected time.  Yes I got the issue, I can also reproduce it.

**Comment by balajis-qb** [comment]
Hi, 
Actually the mentioned scroll issue occurs all the times regardless of the usage of the portalId.  

I fixed the issue in my PR https://github.com/Hacker0x01/react-datepicker/pull/4296 .  Kindly review my fix and let me know your review comments.

**Comment by 1Jesper1** [comment]
> Hi, Actually the mentioned scroll issue occurs all the times regardless of the usage of the portalId.
> 
> I fixed the issue in my PR #4296 . This PR also fixes the other scroll related issue #4289. Kindly review my fix and let me know your review comments.

Thanks for the fast fix for both issues. PR is detailed!
